src/Core/Grid/Filter/FilterFormFactoryFormActionDecorator.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * 2007-2019 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Filter;
+
+use PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinitionInterface;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Decorates grid filter form action.
+ */
+final class FilterFormFactoryFormActionDecorator implements GridFilterFormFactoryInterface
```

Comment:
`FilterFormFactoryFormActionDecorator`
There's an issue with this name 😛:
- twice the "Form" word
- factory and action

I would feel more confortable to have hardcoded `$formActionRoute` because right now this class seems too abstract to me 😟

Reply:
to explain the name - it is a Decorator of FilterFormFactory, which decorates FormAction :sweat_smile:

the purpose of it is to change the filters form action, I couldn't see a better way to do it since it's not configurable besides decorating :thinking:  

the idea was to reuse this decorator, because we already need it in at least two places, ping @zuk3975

Reply:
The base problem - after options form submit, all the `errors are lost between actions`. 
At first i had an idea of `forwarding controller that seemed to work`, but it appeared it has an issue (_leaves `/options` in url after error, and later doesn't accept GET_). So Rokas came up with this idea of decorator which seems to work.

Reply:
To explain further, since there is a grid and options form in the same page, the issue is:
* Grid filters submit route is `/` by default.
* Options need to use `/` submit route, otherwise we will lose form errors as @zuk3975 mentioned. 

It means that we have two different actions with the same path `/` which is invalid. The decorator allows changing the submit route of grid filters to a custom route, leaving the `/` route for options and fixing this issue. :slightly_smiling_face:

Reply:
@matks waiting for your opinion :stuck_out_tongue:

Reply:
You win 😄